cgpsa

```diff
@@ -2909,30 +3004,75 @@ sub readConfigFile
     {
       $conf->{sa_local_rules_dir} = $1;
     }
-    elsif ($line =~ /^absolute[-_]queue[-_]filenames[\s\=]+(\S+)$/) 
+    elsif ($line =~ /^absolute[-_]queue[-_]filenames[\s\=]+(\S+)$/)
     {
       $conf->{absolute_queue_filenames} = evaluateBoolean($1);
     }
     elsif ($line =~ /^prevent[-_]backscatter[\s\=]+(\S+)$/)
     {
       $conf->{prevent_backscatter} = evaluateBoolean($1);
     }
+    elsif ($line =~ /^x[-_]junk[-_]score[-_]header[\s\=]+(\S+)$/)
+    {
+      $conf->{x_junk_score_header} = evaluateBoolean($1);
+    }
+    elsif ($line =~ /^x[-_]junk[-_]score[-_]thresholds[\s\=]+(.+)$/)
+    {
+      # got to parse a space-separated list of floating point numbers;
+      # if there are not 6 of them, or they are not increasing, we
+      # keep the previous settings (which might be defaults)
+
+      my @new_x_junk_score_thresholds = split /\s+/, $1;
+      my $legit_thresholds = 1;
+      my $threshold_count = 0;
+      my $last_threshold = -inf;
+      foreach my $threshold (@x_junk_score_thresholds)
+      {
+        $legit_thresholds =
+          $legit_thresholds && isFloat($threshold) && $last_threshold < $threshold;
+        $last_threshold = $threshold;
+        $threshold_count = $threshold_count + 1;
+      }
+      if ($legit_thresholds && $threshold_count == 6)
```

Comment:
Instead of hard-coding, it should correspond to `@x_junk_score_constants`

Reply:
Yeah, that works. :)